plugins/statistics.koplugin/main.lua

```diff
@@ -2727,27 +2727,14 @@ function ReaderStatistics:onCloseDocument()
     self:insertDB()
 end
 
-function ReaderStatistics:onAddHighlight()
+function ReaderStatistics:onAnnotationsModified(diff)
     if self.settings.is_enabled then
-        self.data.highlights = self.data.highlights + 1
-    end
-end
-
-function ReaderStatistics:onDelHighlight()
-    if self.settings.is_enabled then
-        self.data.highlights = self.data.highlights - 1
-    end
-end
-
-function ReaderStatistics:onAddNote()
-    if self.settings.is_enabled then
-        self.data.notes = self.data.notes + 1
-    end
-end
-
-function ReaderStatistics:onDelNote()
-    if self.settings.is_enabled then
-        self.data.notes = self.data.notes - 1
+        if diff.highlights then
```

Comment:
A bit hard to get into your latest changes and "AnnotationsModified" event's args: an array of tables containing the original annotation, and additional keys as increment. Not pretty at all :/
And I think this caused you to make an error here: `diff` is that array, and you should use `diff[1].highlights`, no?

If we currently don't pass multiple things in that array (and as you seem more keen than me to limit yourself to stuff we currently do :)), just pass as args to the event: the_item, nb_highlights_added, nb_notes_added?
Which I would still find not very pretty.
Was there a reason to pass to statistics the highlights/notes +1/-1, and not just a signal so Statistics can just fetch the correct numbers from RA?

Reply:
This is a table with an array part and a hash part, nothing unusual.

Reply:
I see that we indeed can pass multiple annotation items:
`self.ui:handleEvent(Event:new("AnnotationsModified", { highlight, bookmark_before }))`

But your naming of the counter ("highlights") make it confusing if you compare the above line to:
`self.ui:handleEvent(Event:new("AnnotationsModified", { item, highlights = 1 }))`

May be make them more explicite: `nb_highlights_added = +1/-1`; `nb_notes_added`.

Plus, the arg in the receiver is named differently, which makes it confusing about what it is:
```
ReaderThumbnail:resetCachedPagesForBookmarks(items)
ReaderStatistics:onAnnotationsModified(diff)
```

Reply:
> the arg in the receiver is named differently

To emphasize that we use different parts.
And because of having no idea of the good combined name.

Reply:
Thanks, everything reads better and easier.